dagshub/common/api/repo.py

```diff
@@ -226,6 +231,83 @@ def get_storage_file(self, path: str) -> bytes:
             raise RuntimeError(error_msg)
         return res.content
 
+    def _get_files_in_path(self, path, revision=None, recursive=False) -> List[ContentAPIEntry]:
+        """
+        Walks through the path of the repo, returning non-dir entries
+        """
+
+        dir_queue = [path]
+        files = []
+
+        progress = get_rich_progress(rich.progress.MofNCompleteColumn())
+        task = progress.add_task("Traversing directories...", total=None)
+
+        def step(step_path):
+            res = self.list_path(step_path, revision)
```

Comment:
Will this also work with buckets?
If so, then:
1. It should be mentioned in the `branch` argument that revisions are ignored when downloading from buckets
2. The `:` in the bucket URL prefix needs to be handled correctly, and probably also documented

Reply:
The buckets should be working almost out of the box, with the exception of the `:` symbol not being valid in filenames in windows, so you're right that we'd need to scrub it.

There is a caveat though, and a thing that I should change and it's not hard - the bucket downloader optimization only works on paths that go through `storage/s3/bucket`, and not `src/s3:/bucket`. But that should be easy for me to change from latter to former.

Reply:
Handling the buckets correctly now.
I even managed to do a `repo.download("/", "local/path")` and it even downloaded E V E R Y T H I N G

Let me know if you maybe want me to put auto-downloading storages behind a flag?

Reply:
@kbolashev Great, yeah I think there should be a flag